os/kernel/log_dump/log_dump.c

```diff
@@ -47,11 +54,15 @@
 #define MAX(a, b)		   (((a) > (b)) ? (a) : (b))
 #endif
 
-#define LOG_DUMP_OK			 0
+#define LOG_DUMP_OK			0			/* Used for successful return check */
+#define LOG_DUMP_OK_HEAPCHK		1		/* To check heap available after success return */
 #define LOG_DUMP_MEM_FAIL		-1
 #define LOG_DUMP_HEAP_FAIL		-2
 #define LOG_DUMP_SIZE_FAIL		-3
 #define LOG_DUMP_OPT_FAIL		-4
+#define LOG_DUMP_STR_FREEMEM	14
```

Comment:
What is the difference between `FREEMEM` and `FREEHEAP`?

Reply:
removed unused variables